### PR TITLE
Fix small bugs and add pre- and post-hooks to some roles

### PR DESCRIPTION
**Pull request by wheelerlaw**
I just noticed this in the output:

```
TASK [cleanup : Is project repository present in /home/wlaw/.contra-env-setup/pipeline] ************************************************************************************************************************************************
ok: [localhost]

TASK [cleanup : Cleanup /home/wlaw/.contra-env-setup/pipeline] *************************************************************************************************************************************************************************
skipping: [localhost] => (item=/home/wlaw/.contra-env-setup//home/wlaw/.contra-env-setup/pipeline)
```

You can see in the `Cleanup /home/wlaw/.contra-env-setup/pipeline` task, it is prepending the path with another path, making the path invalid, so it is skipped. 

Also as a part of this MR, the behavior around `start_minishift` was made more consistent, since how that variable was used depended on whether minishift was being provisioned locally or on a remote VM. Now it is more consistent, in that the `start_minishift` variable is respected regardless if the VM is local or remote. The default value for `start_minishift` was changed to `true`, since it is being started anyway by the `os_temps` playbook. 

Finally, pre- and post-hooks were added to some of the playbooks to support adding in custom behavior by any consumers of this project.

### CODE REVIEW

**Comment by wheelerlaw** [comment]
Okay, I also cleaned up some inconsistencies with the startup logic for minishift, and I added some pre- and post-hooks for some of the roles. That way, I can inject the CA cert into minishift when its done being setup but before CVP is deployed.

File: playbooks/roles/minishift/tasks/main.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 ---
 # setup minishift
 
+- name: "Running cleanup pre-tasks {{ minishift_pre_hooks }}"
```
Comment (dirgim) [inline playbooks/roles/minishift/tasks/main.yml:4]: I'm guessing this should be 
```suggestion
- name: "Running minishift pre-tasks {{ minishift_pre_hooks }}"
```

File: playbooks/roles/minishift/tasks/main.yml [hunk 2]
```diff
@@ -74,9 +79,19 @@
...
-
+  block:
+    - name: "Initialize minishift"
+      shell: "{{ minishift_bin }} start --profile {{ profile }} --openshift-version {{ oc_version }} --vm-driver generic --remote-ipaddress {{ minishift_external_vm_ip }} --remote-ssh-user {{ minishift_external_vm_user }} --remote-ssh-key {{ minishift_external_vm_ssh_key_location }}"
+
+    - name: "Stop the minishift profile {{ profile }}"
+      shell: "{{ minishift_bin }} stop --profile {{ profile }}"
+      when: (start_minishift|bool == false)
+
+- name: "Running cleanup post-tasks {{ minishift_post_hooks }}"
```
Comment (dirgim) [inline playbooks/roles/minishift/tasks/main.yml:94]: I'm guessing this should be 
```suggestion
- name: "Running minishift post-tasks {{ minishift_post_hooks }}"
```

**Review by dirgim — CHANGES_REQUESTED** [review]
The changes look pretty good apart from a couple of nits - and since the PR has grown to include more structural changes (additional hooks), I'd ask for the commit and PR titles to be changed to reflect that.

**Comment by wheelerlaw** [comment]
I updated the PR with the requested changes, and also updated the title and description of the PR to reflect the scope of the changes.

**Comment by wheelerlaw** [comment]
Reworded commit message, and squashed all commits into one.

File: playbooks/roles/minishift/tasks/main.yml [hunk 3]
```diff
@@ -74,9 +79,19 @@
   when: ((minishift_profile_status.stdout_lines and minishift_profile_status.stdout_lines[0]|lower == "does not exist") or
         (minishift_profile_status.rc != 0)) and (minishift_external_vm_ip == "")
 
-# Start minishift on an external VM if configured
+# Start minishift on an external VM if configured, then shut it down (if it was configured not to start, but we wanted the profile to exist)
```
Comment (dirgim) [inline playbooks/roles/minishift/tasks/main.yml:82]: Looking into this a bit more, the `minishift profile set` command on [line 25](https://github.com/CentOS-PaaS-SIG/contra-env-setup/pull/175/files#diff-0d002d82212b0a8c719d5df0c6de01c396bbfd48de83b2168ad060c64632106aR25) should already create the profile without the need to start the minishift machine.

Comment (wheelerlaw) [reply playbooks/roles/minishift/tasks/main.yml:82]: Oh interesting, so I suppose I could change it from
- Starting up minishift
- Stopping if `start_minishift == false`
to
- Starting up minishift if `start_minishift == true`?

Comment (dirgim) [reply playbooks/roles/minishift/tasks/main.yml:82]: Yes, I think that should be it,

Comment (wheelerlaw) [reply playbooks/roles/minishift/tasks/main.yml:82]: Okay, I changed the logic

**Comment by wheelerlaw** [comment]
Added some more tweaks, and squashed.

**Comment by wheelerlaw** [comment]
@dirgim Bump

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**